src/run/parser/portal2_live_timer.rs

```diff
@@ -159,18 +156,33 @@ pub fn parse<R: BufRead>(source: R) -> Result<Run> {
 
     let mut aggregate_ticks = 0.0;
 
+    let mut line = lines.next().ok_or(Error::ExpectedMap)??;
     for &(chapter_name, maps) in &CHAPTERS {
         for &map in maps {
-            let line = lines.next().ok_or(Error::ExpectedMap)??;
-            let mut splits = line.split(',');
-            let map_name = splits.next().ok_or(Error::ExpectedMapName)?;
-            if map_name != map {
-                return Err(Error::ExpectedDifferentMapName);
+            {
+                let mut splits = line.split(',');
+                let map_name = splits.next().ok_or(Error::ExpectedMapName)?;
+                if map_name != map {
+                    return Err(Error::ExpectedDifferentMapName);
+                }
+                let start_ticks: f64 = splits.next().ok_or(Error::ExpectedStartTicks)?.parse()?;
+                let end_ticks: f64 = splits.next().ok_or(Error::ExpectedEndTicks)?.parse()?;
+                let map_ticks = end_ticks - start_ticks;
+                aggregate_ticks += map_ticks;
+            }
+
+            while let Some(Ok(next_line)) = lines.next() {
+                line = next_line;
+                let mut splits = line.split(',');
+                if splits.next() == Some(map) {
+                    let start_ticks: f64 = splits.next().ok_or(Error::ExpectedStartTicks)?.parse()?;
+                    let end_ticks: f64 = splits.next().ok_or(Error::ExpectedEndTicks)?.parse()?;
+                    let map_ticks = end_ticks - start_ticks;
+                    aggregate_ticks += map_ticks;
+                } else {
+                    break;
+                }
```

Comment:
I don't like the amount of duplication happening here. Can we turn this into a single loop maybe?